pysmart_exporter/__main__.py

```
@@ -25,6 +28,14 @@ def main():
     registry.register(collector)
     args = collector.args
 
+    if args['sudo']:
```

Instead of checking for a sudo argument (that might be ok, nonetheless) we should check if we are uid=0 and if not just return error with an error messages (use logging instead of print please)

also, if you want to call a subprocess (and keep the capability to get sudo powers) to continue with higher privileges, you should always return its return code instead of 0 plus a message